src/sync/sync-utils.ts

```diff
@@ -49,6 +54,30 @@ export async function findOrStartSync(
 	}, 0, logger);
 }
 
+const resetRepoSyncStateTasksFromSubscription = async (subscription: Subscription, targetTasks?: TaskType[], syncType?: SyncType): Promise<[number, RepoSyncState[]] | undefined> => {
+
+	if (!targetTasks?.length) {
+		return;
+	}
+
+	const updateTasks = {};
+	targetTasks.forEach(task => {
+		if (syncType === "full") {
+			updateTasks[`${task}Cursor`] = null;
+		}
+		updateTasks[`${task}Status`] = null;
+	});
+
+	return RepoSyncState.update({
+		repoUpdatedAt: null,
+		...updateTasks
+	}, {
+		where: {
+			subscriptionId: subscription.id
+		}
+	});
```

Comment:
This completely forgets the `repository` task that's on subscription model (not reposyncstate), but also, it will also try to update `repositoryStatus` which doesn't exist on RepoSyncState and I think will error?

Reply:
ohhh ok, i thought it was on the reposyncstate, but that make sense since its a higher level. copy that.

Reply:
I don't think we need to split this up into extra functions, especially since they have duplicate code in it.  just a simple if statement for `repository` task should do